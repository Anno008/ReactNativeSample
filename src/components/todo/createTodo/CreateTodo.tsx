import styles from "components/todo/createTodo/CreateTodoStyle";
import React from "react";
import { TextInput, View } from "react-native";

interface Props {
    addTodoItem: (todoName: string) => void;
}

interface State {
    todoName: string;
}

export class CreateTodo extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props);
        this.state = {
            todoName: ""
        };
    }

    render(): JSX.Element {
        return (
            <View style={styles.container}>
                <TextInput
                    value={this.state.todoName}
                    onChangeText={this.handleTextChange}
                    onSubmitEditing={this.createTodoItem}
                    placeholder="Enter a new todo"
                    returnKeyType="done"
                />
            </View>
        );
    }

    public handleTextChange = (text: string): void => this.setState({ todoName: text });

    public createTodoItem = (): void => {
        this.props.addTodoItem(this.state.todoName);
        this.setState({ todoName: "" });
    }
}

export default CreateTodo;
