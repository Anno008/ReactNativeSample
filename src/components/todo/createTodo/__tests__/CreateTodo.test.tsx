import CreateTodo from "components/todo/createTodo/CreateTodo";
import { shallow, ShallowWrapper } from "enzyme";
import React from "react";

describe("TodoItem component test", (): void => {
    const mockedProps = {
        addTodoItem: jest.fn()
    };

    it("Should match CreateTodo snapshot", (): void => {
        const wrapper: ShallowWrapper = shallow(<CreateTodo {...mockedProps} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("Should set state when text is changed", (): void => {
        const wrapper: ShallowWrapper<{}, {}, CreateTodo> = shallow(
            <CreateTodo {...mockedProps} />
        );
        const instance = wrapper.instance();

        const setStateSpy = jest.spyOn(instance, "setState");

        instance.handleTextChange("test");

        expect(setStateSpy).toBeCalled();
        expect(setStateSpy).toHaveBeenCalledWith({ todoName: "test" });
    });

    it("Should call addTodoItem and set state when calling createTodoItem", (): void => {
        const wrapper: ShallowWrapper<{}, {}, CreateTodo> = shallow(<CreateTodo {...mockedProps} />);
        const instance = wrapper.instance();

        const todoName = "test todo item";
        instance.setState({ todoName });

        const setStateSpy = jest.spyOn(instance, "setState");
        instance.createTodoItem();

        expect(mockedProps.addTodoItem).toBeCalledWith(todoName);
        expect(setStateSpy).toHaveBeenCalledWith({ todoName: "" });
    });
});
