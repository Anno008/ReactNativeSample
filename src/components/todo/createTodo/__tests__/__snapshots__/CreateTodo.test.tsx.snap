// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem component test should render create todo component 1`] = `ShallowWrapper {}`;
